Lib/test/test_cmd_line.py

```diff
@@ -1051,6 +1051,75 @@ def test_int_max_str_digits(self):
         )
         self.assertEqual(res2int(res), (6000, 6000))
 
+    def test_cmd_dedent(self):
+        # test that -c auto-dedents its arguments
+        from textwrap import dedent
+        test_cases = [
+            {
+                'code': '''
+                    print('space-auto-dedent')
+                ''',
+                'expected': b'space-auto-dedent',
+            },
+            {
+                'code': dedent('''
+                ^^^print('tab-auto-dedent')
+                ''').replace('^', '\t'),
+                'expected': b'tab-auto-dedent',
+            },
+            {
+                'code': dedent('''
+                ^^if 1:
+                ^^^^print('mixed-auto-dedent-1')
+                ^^print('mixed-auto-dedent-2')
+                ''').replace('^', '\t \t'),
+                'expected': b'mixed-auto-dedent-1\nmixed-auto-dedent-2',
+            },
+            {
+                'code': '''
+                    data = """$
+
+                    this data has an empty newline above and a newline with spaces below $
+                                            $
+                    """$
+                    if 1:         $
+                        print(repr(data))$
+                '''.replace('$', ''),
+                # Note: entirely blank lines are normalized to \n, even if they
+                # are part of a data string. This is consistent with
+                # textwrap.dedent behavior, but might not be intuitive.
+                'expected': b"'\\n\\nthis data has an empty newline above and a newline with spaces below \\n\\n'",
+            },
+        ]
+        for case in test_cases:
+            # Run the auto-dedent case
+            args1 = sys.executable, '-c', case['code']
+            proc1 = subprocess.run(args1, stdout=subprocess.PIPE)
+            self.assertEqual(proc1.returncode, 0, proc1)
+            output1 = proc1.stdout.strip()
+
+            # Manually dedent beforehand, check the result is the same.
+            args2 = sys.executable, '-c', dedent(case['code'])
+            proc2 = subprocess.run(args2, stdout=subprocess.PIPE)
+            self.assertEqual(proc2.returncode, 0, proc2)
+            output2 = proc2.stdout.strip()
+
+            self.assertEqual(output1, output2)
+            self.assertEqual(output1.replace(b'\r\n', b'\n'), case['expected'])
```

Comment:
Along with converting the expected test cases into `str`. You could also consider just using a sequence of tuples and iterating over (code, expected) pairs (`for code, expected in cases:`).
```suggestion
            self.assertEqual(output1, case['expected'])
```

Reply:
Resolved